### PR TITLE
[JENKINS-69229] Use newer `trilead-api` on Java 11+ lines

### PR DESCRIPTION
**Pull request by jglick**
See https://github.com/jenkinsci/trilead-api-plugin/pull/55, #1374, #1378, #1379, & #1382.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -10,17 +10,12 @@ updates:
     schedule:
       interval: "daily"
   - package-ecosystem: "maven"
-    open-pull-requests-limit: 10
+    open-pull-requests-limit: 25
```
Comment (jglick) [inline .github/dependabot.yml:13]: https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/327#issuecomment-1288952434

**Review by jtnord — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by jglick** [comment]
Checking to see if https://github.com/jenkinsci/bom/pull/1484/checks?check_run_id=9074926262 is actually caused by this PR’s changes.

**Comment by jglick** [comment]
No, can reproduce also in trunk. Possibly caused by a Git (CLI) update on agents. Will look into this. CC @MarkEWaite

**Review by kuisathaverat — APPROVED** [review]

**Comment by jglick** [comment]
https://github.com/jenkinsci/git-client-plugin/pull/920

**Pull request merged**